src/codegen.rs

```diff
@@ -978,26 +979,43 @@ fn codegen_debug_description_getter(structure: &CheckedStruct, project: &Project
     for (i, field) in structure.fields.iter().enumerate() {
         output.push_str(&codegen_indent(INDENT_SIZE));
 
-        output.push_str("builder.appendff(\"");
-        output.push_str(&field.name);
-        output.push_str(": {}");
-        if i != structure.fields.len() - 1 {
-            output.push_str(", ");
-        }
+        match project.types[field.type_id] {
+            Type::GenericInstance(struct_id, _) => {
+                if struct_id == project.get_optional_struct_id(Span::new(0, 0, 0)) {
```

Comment:
Instead of passing `Span::new(0, 0, 0)` I think you should just pass along the span from `ParsedStruct` to `CheckedStruct`, and then just use that here.

Reply:
fixed